human_analysis/nbconverted/2_identify_generic_genes_pathways.py

```diff
@@ -249,6 +321,144 @@
+
+
+# ## Quick validation
+# Significant pathways found include: HALLMARK_G2M_CHECKPOINT, TNFA_SIGNALING, INFLAMMATORY_RESPONSE, APOPTOSIS, HYPOXIA are expected and consistent with what the publication found. However, there are pathways not found to be significant that we would expect to find: P53, NOTCH_SIGNALING, DNA_REPAIR. Not sure how to explain this since the enrichment score plot looks consistent with the p-value score .
+
+# In[27]:
+
+
+print(template_enriched_pathways.shape)
+template_enriched_pathways[template_enriched_pathways['padj'] < 0.05].sort_values(by='padj')
+
+
+# In[28]:
+
+
+template_enriched_pathways[template_enriched_pathways['padj'] >= 0.05].sort_values(by='padj')
+
+
+# In[29]:
+
+
+template_enriched_pathways.set_index('pathway', inplace=True)
+template_enriched_pathways.loc["HALLMARK_P53_PATHWAY"]
+
+
+# In[30]:
+
+
+# Check expression of TP53 genes in cancer vs normal samples
+tp53_genes = list(template_DE_stats[list(template_DE_stats.index.str.find("TP53") == 0)].index)
+template_data = pd.read_csv(template_data_file, sep="\t", index_col=0)
+
+selected_template_data = template_data[tp53_genes]
+
+
+# In[31]:
+
+
+# Load metadata file with grouping assignments for samples
+metadata_file = os.path.join(
+    "data",
+    "metadata",
+    project_id+"_groups.tsv")
+
+# Read metadata
+metadata = pd.read_csv(
+    metadata_file,
+    header=0,
+    sep='\t',
+    index_col=0)
+
+
+# In[32]:
+
+
+# PCA encode
+pca = PCA(n_components=2)
+
+model = pca.fit(selected_template_data)
+template_PCAencoded = model.transform(selected_template_data)
+
+template_PCAencoded_df = pd.DataFrame(data=template_PCAencoded ,
+                                         index=template_data.index,
+                                         columns=['1','2'])
+
+
+# In[33]:
+
+
+# Add tumor/normal labels
+template_data_labeled = pd.merge(template_PCAencoded_df, metadata, left_index=True, right_index=True)
+template_data_labeled.loc[template_data_labeled['group'] == 1,'source'] = 'Normal'
+template_data_labeled.loc[template_data_labeled['group'] == 2,'source'] = 'Tumor'
+template_data_labeled.head()
+
+
+# In[34]:
+
+
+# Plot
+fig = ggplot(template_data_labeled, aes(x='1', y='2'))
+fig += geom_point(aes(color='source'), alpha=0.7)
+fig += labs(x ='PCA 1',
+            y = 'PCA 2',
+            title = 'PCA template data')
+fig += theme_bw()
+fig += theme(
+    legend_title_align = "center",
+    plot_background=element_rect(fill='white'),
+    legend_key=element_rect(fill='white', colour='white'), 
+    legend_title=element_text(family='sans-serif', size=15),
+    legend_text=element_text(family='sans-serif', size=12),
+    plot_title=element_text(family='sans-serif', size=15),
+    axis_text=element_text(family='sans-serif', size=12),
+    axis_title=element_text(family='sans-serif', size=15)
+    )
+fig += guides(colour=guide_legend(override_aes={'alpha': 1}))
+
+print(fig)
```

Comment:
Are you just checking here to see that the tumor and normal samples separate well?

Reply:
Yes...this was me "looking at the data" to try to rule out what could be causing the lack of p53 significance. So here I was looking to see if my data looked as expected, where the tumor and normal samples separated. The two possible issues I could concieve were that there is an issue in how I processed the data, which is what this PCA plot is validating. Or there is an issue with how I implemented the DE analysis, which also seems to be correct based on the number of DEGs found...so I need to think more about this and perhaps look back at the publication associated with the template experiment.